### PR TITLE
Fix `TokenManagementSpecsStateful` tests

### LINKED ISSUE #8599: Resolve `TokenManagementSpecsStateful` errors
### Problem

We have the following tests suite `TokenManagementSpecsStateful` that has problems when it's running against modularization codebase.

### Solution

Resolve issues with the test suite. When ready add @hapitest annotations to the passing tests.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by natanasow**
**Description**:

The current setup skips `realm` and `shard` in the account id and the returned account with id `0.0.1002` and id `1.2.1002` is the same. Fixed the account id to be really incorrect but it might be better to fix the codebase to validate the realm and shard as well.

**Related issue(s)**:

Fixes #8599 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by povolev15 — DISMISSED** [review]

**Review by netopyr — COMMENTED** [review]
We need to be careful when enabling or changing tests. They need to continue to work with the mono code.

@tinker-michaelj, can you please weight in?

**Comment by natanasow** [comment]
> We need to be careful when enabling or changing tests. They need to continue to work with the mono code.
> 
> @tinker-michaelj, can you please weight in?

Just ran them against the mono repo and they are working as expected. Good to know to not enable more tests and to be more strict with the current ones. Thank :pray: .

**Comment by natanasow** [comment]
@netopyr @tinker-michaelj are we ready to merge this or we're waiting for more approvals?

**Review by tinker-michaelj — CHANGES_REQUESTED** [review]
I think in this case we should keep the test passing as-is.

**Comment by natanasow** [comment]
> I think in this case we should keep the test passing as-is.

Done.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenManagementSpecsStateful.java [hunk 1]
```diff
@@ -65,12 +68,14 @@ public List<HapiSpec> getSpecsInSuite() {
...
     }
 
+    @HapiTest
     public HapiSpec freezeMgmtFailureCasesWork() {
         var unfreezableToken = "without";
         var freezableToken = "withPlusDefaultTrue";
 
         return defaultHapiSpec("FreezeMgmtFailureCasesWork")
                 .given(
+                        cryptoTransfer(tinyBarsFromTo(GENESIS, ADDRESS_BOOK_CONTROL, ONE_HUNDRED_HBARS)),
```
[the remark this answers is not in the document]
Comment (natanasow) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenManagementSpecsStateful.java:78]: Okay.

[the remark this answers is not in the document]
Comment (natanasow) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenManagementSpecsStateful.java:78]: Did you double-check it?

[the remark this answers is not in the document]
Comment (netopyr) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenManagementSpecsStateful.java:78]: Yes, privileged transactions like this one are free. It should not be needed to transfer Hbar to `ADDRESS_BOOK_CONTROL`. One of the PRs I reviewed recently contained a fix. Not sure which one, though. 🙂 But I think I marked you there.

[the remark this answers is not in the document]
Comment (natanasow) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenManagementSpecsStateful.java]: I didn't receive any notification/email :confused:. Reverted the change, pulled develop into this branch and all tests are green now, so we are ready for merge, right?

**Review by netopyr — APPROVED** [review]
LGTM

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenManagementSpecsStateful.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (netopyr) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenManagementSpecsStateful.java]: Right. As soon as @tinker-michaelj unblocks the PR, we can merge it.

**Pull request merged**